test/blackbox-tests/test-cases/exec-watch/exec-watch-basic.t

```diff
@@ -0,0 +1,52 @@
+Test exec --watch with a program that terminates immediately.
+
+  $ cat >dune-project <<EOF
+  > (lang dune 3.6)
+  > (package (name foo))
+  > EOF
+
+  $ cat >dune <<EOF
+  > (executable
+  >  (public_name foo))
+  > EOF
+
+  $ cat >foo.ml <<EOF
+  > let () = print_endline "foo"
+  > EOF
+
+  $ dune exec --watch ./foo.exe &
+  Success, waiting for filesystem changes...
+  foo
+  Success, waiting for filesystem changes...
+  bar
+  File "foo.ml", line 1, characters 23-24:
+  1 | let () = print_endline "baz
+                             ^
+  Error: String literal not terminated
+  Had errors, waiting for filesystem changes...
+  Success, waiting for filesystem changes...
+  baz
+  $ PID=$!
+
+  $ sleep 1
+
+  $ cat >foo.ml <<EOF
+  > let () = print_endline "bar"
+  > EOF
+
+  $ sleep 1
+
+  $ cat >foo.ml <<EOF
+  > let () = print_endline "baz
+  > EOF
+
+  $ sleep 1
+
+  $ cat >foo.ml <<EOF
+  > let () = print_endline "baz"
+  > EOF
+
+  $ sleep 1
```

Comment:
This test is a bit unreliable (and slow) for my tastes. Can you use `--passive-watch-mode` and manually trigger the watch mode notifications instead of sleeping for them? There should be examples in the test suite of this.

Reply:
I haven't implemented passive-watch-mode for exec (only eager watch mode). I could add it, but it won't totally solve the problem that I'm (admittedly poorly) solving with `sleep` at the moment. If I'm correctly understanding how passive watch mode works for the build command, the rpc client can be made to block until the build has finished. For exec, the equivalent would be to block until the command finishes running, but that's not easy to do without restructuring this change. I think we'd need to start a new thread which is dedicated to waiting for the spawned process to terminate, then reply to the rpc.

I played around a bit today sending the output of exec to a fifo and draining the fifo at various points in the test but I don't have a good solution yet. There's also the question of what to do about the `exec-watch-server.t` test which runs a program that blocks forever to simulate a server. The point of this test is that the executed command never completes and dune has to kill the child process before starting another, and it's not clear to me what the event should be that triggers the test to proceed.

I suppose we could have the exec'd program itself communicate with the test in some way. @rgrinberg do you think that sounds reasonable?

Reply:
Yes, that sounds reasonable to me. Anything that avoids timing issues is good enough.

Reply:
Ok instead of sleeping, I've updated it so that the exec'd program touches a file to indicate that it's finished, and the test waits for the file to exist before proceeding.